oeps/oep-0045/decisions/0001-tutor-as-replacement-for-edx-configuration.rst

```diff
@@ -0,0 +1,328 @@
+<https://discuss.openedx.org/t/deploying-mfes-in-the-community/2868/22>`__), this shouldn't count against it. In any
+case, the project leadership is `aware of the issue
+<https://discuss.overhang.io/t/work-in-progress-upgrading-tutor-to-koa/1071>`__.
+
+
+External databases
+``````````````````
+
+An important characteristic of scaling up an Open edX deployment is having the ability to point it to external
+databases, such as Amazon RDS or a manually maintained Galera cluster. Tutor allows for just this. All of the following
+can be disabled locally and easily `configured to use an externally hosted service
+<https://docs.tutor.overhang.io/configuration.html#mysql>`__:
+
+-  MySQL
+-  Elasticsearch
+-  MongoDB
+-  Redis
+-  SMTP
+
+
+Multi-server
+````````````
+
+Single-server deployments are less than ideal in terms of high availability and scalability, so for larger deployments
+something different is needed. And given that Tutor deployments use Docker images, Kubernetes would be the first choice.
+(Not least of which because Kubernetes is also what edX `has started to use
+<https://discuss.openedx.org/t/lets-talk-about-the-native-installation/3269/14>`__.)
+
+Tutor does come with basic support for Kubernetes deployment.  ``tutor k8s quickstart`` on a local `MicroK8s
+<https://microk8s.io/>`__ 1.8 installation did not work out of the box, though. There is a race condition during the
+startup of the MySQL container that requires relaunching the command, and one has to modify the MinIO hostname manually
+to avoid a failure during the initial LMS job. There were further MinIO name resolutions issues that precluded a fully
+working environment: there was no time to ascertain whether the cause is ultimately with Tutor or with the tested
+microk8s setup.
+
+It is mentioned at the top of the `Tutor Kubernetes documentation <https://docs.tutor.overhang.io/k8s.html>`__ that "the
+local Open edX install was designed such that users with no prior experience with system administration could still
+launch an Open edX platform. It is not the case for the installation method outlined here." While the Tutor project has
+clearly expended effort in getting it to work with Kubernetes, if it is to fully replace ``edx/configuration`` as the
+go-to community installation for production use, it is suggested that the Kubernetes documentation be extended with more
+detailed information on how to run and maintain Open edX in such an environment, and that Kubernetes support be made a
+first-class citizen.
+
+Note that it is perfectly possible to bypass Tutor entirely when it comes to cluster management. One could use its image
+creation engine to build the requisite Docker images, then manage the Kubernetes cluster with something like Helm: there
+are reports of `this exact thing being done
+<https://discuss.openedx.org/t/lets-talk-about-the-native-installation/3269/2>`__).
```

Comment:
> shouldn't tutor k8s just be removed?

I have thought about it; but for many users, having a working (albeit less user-friendly) K8s deployment is an insurance policy for the future, so I decided to keep k8s deployment. It might move to a separate plugin in the future, but I will keep supporting it.

Reply:
I agree that `tutor k8s` should remain.  What I'm arguing for is to add to it, either in the form of more configurability, or simply documenting an alternative.  I imagine, for example, we could contribute Helm charts and accompanying docs and they would live in `/contrib`, or something like that.